p8e-api/src/main/kotlin/io/provenance/engine/event/IndexHandler.kt

```diff
@@ -169,4 +182,34 @@ class IndexHandler(
         request.opType(OpType.INDEX)
         return request
     }
+
+    private fun updateDataAccess(envelope: EnvelopeRecord) {
+        val envelopeDataAccess = envelope.data.input.affiliateSharesList.map {
+            affiliateSharePublicKey ->
+            affiliateService.getAddress(
+                affiliateSharePublicKey.toPublicKey(), chaincodeProperties.mainNet
+            )
+        }
+        val existingScopeDataAccess = provenanceGrpcService.retrieveScopeData(envelope.data.input.scope.uuid.value).scope.scope.dataAccessList
+        // Only perform job if data access will be updated
+        if (envelopeDataAccess.any { it !in existingScopeDataAccess }) {
+            p8e.Jobs.MsgAddScopeDataAccessRequest.newBuilder()
+                .addAllDataAccess(envelopeDataAccess)
+                .addAllSigners(envelope.data.result.signaturesList.map {
```

Comment:
By any chance did you test this with a multiparty contract? I'm not positive, but I think the only signer we want is the scope owner. In the case for our scopes that's always a single invoker. If you only ran single party contracts, the `envelope.data.result.signaturesList` should only contain the invoker so it's not a big deal. It would take a multiparty contract to see if this fails or not as is.